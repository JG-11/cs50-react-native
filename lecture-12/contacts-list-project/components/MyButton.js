import React from 'react'
import {Â Button } from 'react-native'

export default () => <Button title="Test button" onPress={() => {}} />
