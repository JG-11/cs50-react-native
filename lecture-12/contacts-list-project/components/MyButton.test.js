import React from 'react'
import renderer from 'react-test-renderer'
import MyButton from './MyButton'

describe('MyButton', () => {
  it('renders', () => {
    const button = renderer.create(<MyButton />).toJSON()
    expect(button).toMatchSnapshot()
  })
  
  it('correctly overrides default color', () => {
    const color = 'blue'
    const button = renderer.create(<MyButton color={color}/>).root
    expect(button.props.color).toBe(color)
  })
})
