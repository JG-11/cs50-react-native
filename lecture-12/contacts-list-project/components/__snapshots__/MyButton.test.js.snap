// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyButton renders 1`] = `
<View
  accessibilityRole="button"
  accessibilityStates={Array []}
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#007AFF",
            "fontSize": 18,
            "padding": 8,
            "textAlign": "center",
          },
        ]
      }
    >
      Test button
    </Text>
  </View>
</View>
`;
